backend/src/main/java/com/yigongil/backend/domain/message/MessageRepository.java

```diff
@@ -0,0 +1,31 @@
+package com.yigongil.backend.domain.message;
+
+import com.yigongil.backend.domain.member.Member;
+import java.util.List;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Slice;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.Repository;
+
+public interface MessageRepository extends Repository<Message, Long> {
+
+    Message save(Message message);
+
+    @Query("""
+                        select m.sender as sender, m.receiver as receiver
+                        from Message m
+                        group by m.sender, m.receiver
+                        having m.sender = :member or m.receiver = :member
```

Comment:
지금 쿼리는 다음과 같이 message 테이블을 풀스캔하네요.(type = all)

<img width="900" alt="image" src="https://github.com/woowacourse-teams/2023-yigongil/assets/74969077/ee673cbe-8a51-4982-af51-32e8099b04ea">

where 절을 사용해 수정하면 type = index merge 가 되어 필요한 컬럼만 효율적으로 스캔할 수 있습니다! 👍

<img width="790" alt="image" src="https://github.com/woowacourse-teams/2023-yigongil/assets/74969077/e11c0487-bdab-43c7-8307-eac5b2d98efb">

추가로 이 정보도 페이징 할 것일면 order 절을 넣어주면 좋겠네요! (메서드 이름에 있는데 쿼리엔 없어요)

```suggestion
SELECT m.sender AS sender, m.receiver AS receiver
FROM Message m
WHERE m.sender = :member OR m.receiver = :member
GROUP BY m.sender, m.receiver
```

Reply:
좋은데요? 그리고 unique 값을 뽑아내기 위해 group by를 쓰셨다면 distinct를 쓰시는 걸 추천드립니다. 
성능도 더 좋고 의미적으로도 더 잘 전달될 것 같아요

Reply:
추가로 인덱스 머지 최적화에 대해 가볍게 정리해봤는데 혹시 도움이 될까 싶어 [링크](https://github.com/poi1649/learning/blob/main/db/Index%20Merge%20Optimization.md) 추가합니다.

Reply:
좋습니다